API/NoDocs/Microsoft.AspNetCore.Telemetry/LoggingOptions.cs

```diff
@@ -0,0 +1,39 @@
+// Assembly 'Microsoft.AspNetCore.Telemetry.Middleware'
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
+using Microsoft.Extensions.Compliance.Classification;
+using Microsoft.Extensions.Http.Telemetry;
+using Microsoft.Shared.Data.Validation;
+
+namespace Microsoft.AspNetCore.Telemetry;
+
+public class LoggingOptions
+{
+    public bool LogRequestStart { get; set; }
+    public bool LogBody { get; set; }
+    public IncomingPathLoggingMode RequestPathLoggingMode { get; set; }
+    [Experimental("EXTEXP0003", UrlFormat = "https://aka.ms/dotnet-extensions-warnings/{0}")]
+    public HttpRouteParameterRedactionMode RequestPathParameterRedactionMode { get; set; }
+    [TimeSpan(1, 60000)]
+    public TimeSpan RequestBodyReadTimeout { get; set; }
+    [Range(1, 1572864)]
+    public int BodySizeLimit { get; set; }
+    [Required]
+    public IDictionary<string, DataClassification> RouteParameterDataClasses { get; set; }
```

Comment:
I think this was discussed but I can't find it either -- we generally return concrete collection types in core library API's I think. There's been mention of a principle "be lenient in what you accept and specific in what you return" but more concretely, doing this means that callers essentially can't use any API on Dictionary that isn't on IDictionary (the latter being increasingly smaller than the former). Worst case, they type check, or copy into a Dictionary, or just down cast and then we break them later. 

(Famously Environment.GetEnvironmentVariables() has always returned IDictionary, and internally that was a Hashtable. Then later we changed to returning a Dictionary and broke customers doing downcasts, so we changed back to Hashtable and thus it remains for ever. of course, Dictionary wasn't available when this was written, but if it had explicitly returned Hashtable in the first place we wouldn't have broken anyone.)